sphinx/ext/viewcode.py

```diff
@@ -49,20 +49,41 @@ class viewcode_anchor(Element):
 
 def _get_full_modname(modname: str, attribute: str) -> str | None:
     try:
-        if modname is None:
-            # Prevents a TypeError: if the last getattr() call will return None
-            # then it's better to return it directly
-            return None
-        module = import_module(modname)
-
-        # Allow an attribute to have multiple parts and incidentally allow
-        # repeated .s in the attribute.
-        value = module
-        for attr in attribute.split('.'):
-            if attr:
-                value = getattr(value, attr)
+        try:
+            if modname is None:
+                # Prevents a TypeError: if the last getattr() call will return None
+                # then it's better to return it directly
+                return None
+            module = import_module(modname)
+
+            # Allow an attribute to have multiple parts and incidentally allow
+            # repeated .s in the attribute.
+            value = module
+            for attr in attribute.split("."):
+                if attr:
+                    value = getattr(value, attr)
+
+            return getattr(value, "__module__", None)
+        except ModuleNotFoundError:
+            # Attempt to find full path of module
+            module_path = modname.split(".")
+            actual_path = __import__(module_path[0], globals(), locals(), [], 0)
```

Comment:
It doesn't feel great to use ``__import__()``, and it is explicitly discouraged.

Would [``find_spec()``](https://docs.python.org/3/library/importlib.html#checking-if-a-module-can-be-imported) be suitable?

Reply:
@AA-Turner `find_spec` seems to work in place of `__import__` based on the unit tests for the issue being successful.

With the import path being: `main_package.subpackage.submodule`, and the code generating the correct directory path of: `main_package.subpackage._subpackage2.submodule`